node/index.js

```diff
@@ -582,6 +582,12 @@ TChannelConnection.prototype.onReaderError = function onReaderError(err) {
         localName: self.channel.hostPort,
         error: err
     });
+
+    // TODO instead of resetting send an error frame first.
+    // and reset the socket after sending an error frame
+    self.resetAll(new Error('tchannel read error'));
```

Comment:
why not pass in a wrapped error and remove the log site above (doesn't resetAll do an error log for us?)